### PR TITLE
[REF] *: convert legacy dialogs

### PR DESCRIPTION
**Pull request by FrancoisGe**
[REF] *: convert legacy dialogs

The goal of this commit is to convert a bunch of legacy dialogs into owl dialogs.

TaskID: 3453920

Description of the issue/feature this PR addresses:

Current behavior before PR:

Desired behavior after PR is merged:

---
I confirm I have signed the CLA and read the PR guidelines at www.odoo.com/submit-pr

### CODE REVIEW

File: addons/payment/static/src/xml/payement_manage_form_confirmation_dialog.xml [hunk 1]
```diff
@@ -0,0 +1,18 @@
+<?xml version="1.0" encoding="UTF-8"?>
+
+<templates xml:space="preserve">
+    <t t-name="payement.ManageFormConfirmationDialog" owl="1">
```
Comment (Feyensv) [inline addons/payment/static/src/xml/payement_manage_form_confirmation_dialog.xml:4]:
```suggestion
    <t t-name="payment.ManageFormConfirmationDialog" owl="1">
```
payement -> payment in file name as well.

File: addons/payment/static/src/xml/payement_manage_form_confirmation_dialog.xml [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+                <span>It is currently linked to the following documents:</span>
+                <ul>
+                    <t t-foreach="linkedRecordsInfo" t-as="documentInfo" t-key="documentInfo_index">
+                        <li t-att-title="documentInfo.description" t-att-href="documentInfo.url" t-esc="documentInfo.name"/>
+                    </t>
+                </ul>
+            </t>
+        </div>
+    </t>
+</templates>
```
Comment (Feyensv) [inline addons/payment/static/src/xml/payement_manage_form_confirmation_dialog.xml:18]: There is a bunch of files you are adding without a blank newline.
Might be worth using configuring your editor to automatically fix those.

File: addons/payment/static/src/js/payment_form_mixin.js [hunk 3]
```diff
@@ -70,18 +69,14 @@
...
             const $checkedRadios = this.$('input[name="o_payment_radio"]:checked');
             if ($checkedRadios.length !== 1) { // Cannot find selected payment option, show dialog
-                return new Dialog(null, {
+                this.call("dialog", "add", ConfirmationDialog, {
                     title: sprintf(_t("Error: %s"), title),
-                    size: 'medium',
-                    $content: `<p>${escape(description) || ''}</p>`,
-                    buttons: [{text: _t("Ok"), close: true}]
-                }).open();
+                    body: escape(description) || '',
```
Comment (sdegueldre) [inline addons/payment/static/src/js/payment_form_mixin.js:78]: probably shouldn't be escaped

File: addons/portal/static/src/js/components/input_confirmation_dialog/input_confirmation_dialog.js [hunk 4]
```diff
@@ -0,0 +1,47 @@
...
+                    this.inputEl.focus();
+                    this.inputEl.addEventListener("keydown", onKeydown);
+                    this.inputEl.addEventListener("input", onInput);
+                    return () => {
+                        this.inputEl.removeEventListener("keydown", onKeydown);
+                        this.inputEl.removeEventListener("input", onInput);
+                    };
+                }
+            },
+            () => [this.modalRef.el]
```
Comment (sdegueldre) [inline addons/portal/static/src/js/components/input_confirmation_dialog/input_confirmation_dialog.js:40]: should depend directly on the inputEl: `this.modalRef.el?.querySelector("input")`

File: addons/portal/static/src/js/portal_security.js [hunk 5]
```diff
@@ -199,65 +173,52 @@ publicWidget.registry.RevokeSessionsButton = publicWidget.Widget.extend({
  *
  * @param {Function} rpc Widget#_rpc bound do the widget
  * @param {Promise} wrapped promise to check for an identity check request
+ * @param {Function} call Widget#call bound do the widget
  * @returns {Promise} result of the original call
  */
-export function handleCheckIdentity(rpc, wrapped) {
+export async function handleCheckIdentity(rpc, wrapped, call) {
```
Comment (sdegueldre) [inline addons/portal/static/src/js/portal_security.js:179]: rename call to `addDialog` and pass it `this.call.bind(this, "dialog", "add")`

File: addons/portal/static/src/xml/portal_security.xml [hunk 6]
```diff
@@ -39,22 +39,17 @@
...
     <!-- Popup's view !-->
-    <t t-name="portal.revoke_all_devices_popup_template">
+    <t t-name="portal.revoke_all_devices_popup_template" owl="1">
         <section>
             <div>
                 You are about to log out from all devices that currently have access to this user's account.
             </div><br/>
-            <t t-set="path">password</t>
             <form action="/my/security" method="post" class="oe_reset_password_form" >
-                <input type="hidden" name="op" value="revoke"/>
```
Comment (sdegueldre) [inline addons/portal/static/src/xml/portal_security.xml:49]: add back?

File: addons/portal/static/src/js/components/input_confirmation_dialog/input_confirmation_dialog.js [hunk 7]
```diff
@@ -0,0 +1,47 @@
...
+        );
+    }
+
+    get context() {
```
Comment (sdegueldre) [inline addons/portal/static/src/js/components/input_confirmation_dialog/input_confirmation_dialog.js:44]: remove context

File: addons/web/static/src/core/confirmation_dialog/confirmation_dialog.js [hunk 8]
```diff
@@ -12,46 +12,58 @@ export class ConfirmationDialog extends Component {
...
-        if (this.props.confirm) {
+        if (callback) {
             try {
-                await this.props.confirm();
+                const isProcess = await callback(this.context);
+                if (isProcess === false) {
+                    this.isConfirmedOrCancelled = false;
+                    this.enableButtons();
+                    return;
+                }
```
Comment (sdegueldre) [inline addons/web/static/src/core/confirmation_dialog/confirmation_dialog.js:59]: move this out of try/catch

File: addons/web/static/src/core/confirmation_dialog/confirmation_dialog.js [hunk 9]
```diff
@@ -12,46 +12,58 @@ export class ConfirmationDialog extends Component {
...
-    async _confirm() {
+
+    enableButtons() {
+        if (!this.modalRef.el) {
+            return; // safety belt for stable versions
+        }
+        for (const button of [...this.modalRef.el.querySelectorAll(".modal-footer button")]) {
+            button.disabled = false;
+        }
+    }
```
Comment (sdegueldre) [inline addons/web/static/src/core/confirmation_dialog/confirmation_dialog.js:44]: fuse these and do setButtonsDisabled(true/flase). You can also remove the safety belt for stable

File: addons/web/static/src/core/confirmation_dialog/confirmation_dialog.js [hunk 10]
```diff
@@ -12,46 +12,58 @@ export class ConfirmationDialog extends Component {
...
         if (this.isConfirmedOrCancelled) {
             return;
         }
         this.isConfirmedOrCancelled = true;
         this.disableButtons();
-        if (this.props.confirm) {
+        if (callback) {
             try {
-                await this.props.confirm();
+                const isProcess = await callback(this.context);
```
Comment (sdegueldre) [inline addons/web/static/src/core/confirmation_dialog/confirmation_dialog.js:54]: shouldClose

File: addons/web_editor/static/src/js/common/ace.js [hunk 11]
```diff
@@ -691,13 +692,15 @@ var ViewEditor = Widget.extend({
...
-            Dialog.alert(self, '', {
-                title: _t("Server error"),
-                $content: $('<div/>').html(
-                    _t("A server error occured. Please check you correctly signed in and that the file you are saving is correctly formatted.")
-                    + '<br/>'
-                    + error
-                )
+            const body = markup(`
+                <div>
+                    ${_t("A server error occured. Please check you correctly signed in and that the file you are saving is correctly formatted.")}
```
Comment (sdegueldre) [inline addons/web_editor/static/src/js/common/ace.js:697]: doesn't work

File: addons/web_editor/static/src/js/common/ace.js [hunk 12]
```diff
@@ -691,13 +692,15 @@ var ViewEditor = Widget.extend({
...
+                </div>
+                <br/>
+                ${error}}
```
Comment (sdegueldre) [inline addons/web_editor/static/src/js/common/ace.js:700]: extra closing brace

File: addons/website_forum/static/src/components/flag_mark_as_offensive/flag_mark_as_offensive.js [hunk 13]
```diff
@@ -0,0 +1,32 @@
+/** @odoo-module **/
+
+import { Component, useEffect } from "@odoo/owl";
+import { useChildRef } from "@web/core/utils/hooks";
+import { Dialog } from "@web/core/dialog/dialog";
+
+export class FlagMarkAsOffensiveDialog extends Component {
```
Comment (sdegueldre) [inline addons/website_forum/static/src/components/flag_mark_as_offensive/flag_mark_as_offensive.js:7]: FlagAsOffensive

File: addons/website_forum/static/src/components/flag_mark_as_offensive/flag_mark_as_offensive.js [hunk 14]
```diff
@@ -0,0 +1,32 @@
...
+                if (modalEl) {
+                    const discardButton = modalEl.querySelector(".btn-link");
+                    discardButton.addEventListener("click", onClickDiscard);
+                    return () => {
+                        discardButton.removeEventListener("click", onClickDiscard);
+                    };
+                }
+            },
+            () => [this.modalRef.el]
+        );
```
Comment (sdegueldre) [inline addons/website_forum/static/src/components/flag_mark_as_offensive/flag_mark_as_offensive.js:30]: useExternalListener

File: addons/website_forum/static/src/components/flag_mark_as_offensive/flag_mark_as_offensive.js [hunk 15]
```diff
@@ -0,0 +1,32 @@
...
+            (modalEl) => {
+                if (modalEl) {
+                    const discardButton = modalEl.querySelector(".btn-link");
+                    discardButton.addEventListener("click", onClickDiscard);
+                    return () => {
+                        discardButton.removeEventListener("click", onClickDiscard);
+                    };
+                }
+            },
+            () => [this.modalRef.el]
```
Comment (sdegueldre) [inline addons/website_forum/static/src/components/flag_mark_as_offensive/flag_mark_as_offensive.js:29]: `this.modalRef.el?.querySelector`

File: addons/website_forum/static/src/components/flag_mark_as_offensive/flag_mark_as_offensive.js (hunk 13 above)
Comment (flch-odoo) [reply addons/website_forum/static/src/components/flag_mark_as_offensive/flag_mark_as_offensive.js:7]: No, the post is here already flagged and this is used to convert it to offensive.
Maybe `convertFlaggedToOffensive` if you do want to rename it.

Comment (sdegueldre) [reply addons/website_forum/static/src/components/flag_mark_as_offensive/flag_mark_as_offensive.js:7]: My mistake, seemed to be redundant at first glance.

**Review by AntoineVDV — APPROVED** [review]
Kudos for updating docstrings :+1:

**Comment by FrancoisGe** [comment]
Hello @odoo/rd-security :)
We need a security review:
We convert few Legacy Dialog to Owl Dialog so we just move some code. Like:
- window.location = window.location: to reload the window without submit the form again
- Markup html create by the backend.
- window.location.href = … Redirect to static URL

**Comment by mart-e** [comment]
@robodoo override=ci/security
Thanks !

**Comment by sdegueldre** [comment]
@robodoo r+

**Comment by sdegueldre** [comment]
@robodoo r+

**Pull request closed without merging**